### PR TITLE
MF-517: hide hamburger menu on desktop

### PR DESCRIPTION
**Pull request by seankwon**
# Description

Hides the sidebar and hamburger menu for all pages when view is not desktop

### tablet
Screen Shot 2021-04-01 at 10 17 40 PM

### desktop
Screen Shot 2021-04-01 at 10 17 34 PM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -59,14 +59,14 @@
     "@babel/preset-react": "^7.10.4",
     "@babel/preset-typescript": "^7.10.4",
     "@babel/runtime": "^7.12.13",
-    "@openmrs/esm-framework": "next",
+    "@openmrs/esm-framework": "^3.1.10-alpha.0",
```
Comment (seankwon) [inline package.json]: sorry, need to change this. dunno know how that happened.

File: src/components/navbar-header-panels/side-menu-panel.component.tsx [hunk 2]
```diff
@@ -1,7 +1,11 @@
 import React from "react";
 import { SideNav } from "carbon-components-react/es/components/UIShell";
 import { SideNavProps } from "carbon-components-react";
-import { ExtensionSlot } from "@openmrs/esm-framework";
+import {
+  ExtensionSlot,
+  useExtensionSlot,
```
Comment (seankwon) [inline src/components/navbar-header-panels/side-menu-panel.component.tsx]: must remove.

File: src/components/navbar/navbar.component.tsx [hunk 3]
```diff
@@ -68,12 +73,14 @@ const Navbar: React.FC<NavbarProps> = ({
         <HeaderContainer
           render={({ isSideNavExpanded, onClickSideNavExpand }) => (
             <Header aria-label="OpenMRS">
-              <HeaderMenuButton
-                aria-label="Open menu"
-                isCollapsible
-                onClick={() => togglePanel("sideMenu")}
-                isActive={isActivePanel("sideMenu")}
-              />
+              {layout === "tablet" && (
```
Comment (seankwon) [inline src/components/navbar/navbar.component.tsx]: brb, i need a helper func for this to deal with layout = phone too.

File: src/components/navbar/useNavbarListener.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import React, { useState, useEffect } from "react";
```
Comment (seankwon) [inline src/components/navbar/useNavbarListener.ts]: must remove this.

File: src/root.component.tsx [hunk 5]
```diff
@@ -2,7 +2,11 @@ import React from "react";
 import styles from "./root.styles.css";
 import Navbar from "./components/navbar/navbar.component";
 import { BrowserRouter, Redirect } from "react-router-dom";
-import { getCurrentUser, createErrorHandler } from "@openmrs/esm-framework";
+import {
+  getCurrentUser,
+  createErrorHandler,
+  useExtensionStore
```
Comment (seankwon) [inline src/root.component.tsx]: must remove this.

**Review by harshit151001 — COMMENTED** [review]
Looks good to me @seankwon

File: src/utils.ts [hunk 6]
```diff
@@ -0,0 +1 @@
+export const isTablet = layout => layout === "tablet" || layout === "phone";
```
Comment (FlorianRappl) [inline src/utils.ts]: `layout !== 'desktop'`? Would be also more accurate to have `isDesktop` and do `!isDesktop(...)` or?

Comment (seankwon) [reply src/utils.ts]: yea, that's simpler. will update.

Comment (seankwon) [reply src/utils.ts]: @FlorianRappl https://github.com/openmrs/openmrs-esm-primary-navigation/pull/65/commits/5c9cc33dec48308a60f5fe331450239886ec77f2 updated to use isDesktop

**Review by FlorianRappl — APPROVED** [review]
LGTM! (Must fix the merge conflicts)

**Pull request merged**